std/json.d

```diff
@@ -346,10 +346,11 @@ struct JSONValue
             type_tag = JSON_TYPE.STRING;
             store.str = arg;
         }
-        else static if(is(T : char[]))
+        else static if(is(T : const char[]) || is(T : const wchar[]) || is(T : const dchar[]))
         {
+            import std.utf : byUTF;
             type_tag = JSON_TYPE.STRING;
-            store.str = arg.idup;
+            store.str = byUTF!char(arg).array.idup;
```

Comment:
`isSomeString` from std.traits will take everything. So to exclude what's always been handled:

```D
static if (isSomeString!T &&  !is(T==string))
```